cypress/support/TestListManager.js

```diff
@@ -3,225 +3,140 @@
+  
+  var testGroup = GetTestGroup(testGroupName);
+  if (!testGroup)
   {
-    name: 'Tools', tests:
-    [
-      { name: 'Visit Home Page', func: tools.VisitHomePage},
-      { name: 'Create Model 1', func: tools.CreateModel1},
-      { name: 'Create Model 2', func: tools.CreateModel2},
-      { name: 'Create Model 3', func: tools.CreateModel3},
-      { name: 'Create Model 4', func: tools.CreateModel4},
-      { name: 'Create Model 5', func: tools.CreateModel5},
-    ]
-  },
-]
-
-export function AddToCypressTestList(testList) 
-{
-  var funcName = `AddToCypressTestList()`
-  helpers.ConLog(funcName, `List of Tests: ${testList}`)
+    testGroup = { name: testGroupName, tests: new Array() };
+    testGroups.push(testGroup);
+  }
+
+  var test = { name: testDescription, func: testFunction };
+  testGroup.tests.push(test);
+
+  var testSpecification = `${testGroupName}.${testFunctionName}`;
+  if (-1 == testLists.masterListOfAllTestCases.indexOf(testSpecification)) {
+    throw `There is a syncronization error between our master test list and a TestCase specification for: '${testSpecification}'` }
+}
+
+
+export function AddToCypressTestList(testList) {
+  var funcName = `AddToCypressTestList()`;
+  helpers.ConLog(funcName, `List of Tests: ${testList}`);
   
-  if (!Array.isArray(testList)) testList = [testList]
+  if (!Array.isArray(testList)) testList = [testList];
   
-  var testListIterator = new TestListIterator(testList)
+  var testListIterator = new TestListIterator(testList);
   
-  var test = testListIterator.next
+  var test = testListIterator.next;
   while (test != undefined)
   {
-    helpers.ConLog(funcName, `Adding Group: ${test.group}`)
-    describe(test.group, () =>
-    {
-      var currentGroupName = test.group
+    helpers.ConLog(funcName, `Adding Group: ${test.group}`);
+    describe(test.group, () => {
+      var currentGroupName = test.group;
       while (test != undefined && test.group == currentGroupName)
       {
-        helpers.ConLog(funcName, `Adding Test Case: ${test.name}`)
-        it(test.name, test.func)
-        test = testListIterator.next
+        helpers.ConLog(funcName, `Adding Test Case: ${test.name}`);
+        it(test.name, test.func);
+        test = testListIterator.next;
       }
     })
   }
 }
 
-class TestListIterator
-{
-  constructor(testList)
-  {
-    this.testList = testList
-    this.index = 0
-    this.currentGroup = {name: ''}
+class TestListIterator {
+  constructor(testList) {
+    this.testList = testList;
+    this.index = 0;
+    this.currentGroup = {name: ''};
   }
 
   // groupName.testName - 'testName' from 'groupName'
   // TODO: Add support for these wild card versions
   // *.*                - All Groups, All Tests
   // *.testName         - All tests with all groups matching 'testName'
   // groupName.*        - All tests from 'groupName'
-  get next()
-  {
-    if (this.index >= this.testList.length) return undefined
+  get next() {
```

Comment:
Given that there is already an iterator protocol and convention it's confusing to invest yet another.

I'd at least stick with `next()` as a regular function instead of a getter method.

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator/next

Reply:
Originally I had used another technique for the iterator, one I found online for this pattern, however, it did not work with Cypress for some reason and I spent many hours trying to figure out why. For this PR, I will check this code in as is. I will take another look at the JavaScript iterator pattern and try once again to make it work. At a minimum I'll convert "next" to a regular function as requested.